internal/fs/vss_windows.go

```diff
@@ -0,0 +1,986 @@
+	setContext                    uintptr
+	startSnapshotSet              uintptr
+	addToSnapshotSet              uintptr
+	doSnapshotSet                 uintptr
+	deleteSnapshots               uintptr
+	importSnapshots               uintptr
+	breakSnapshotSet              uintptr
+	getSnapshotProperties         uintptr
+	query                         uintptr
+	isVolumeSupported             uintptr
+	disableWriterClasses          uintptr
+	enableWriterClasses           uintptr
+	disableWriterInstances        uintptr
+	exposeSnapshot                uintptr
+	revertToSnapshot              uintptr
+	queryRevertStatus             uintptr
+}
+
+// getVTable returns the vtable for IVssBackupComponents.
+func (vss *IVssBackupComponents) getVTable() *IVssBackupComponentsVTable {
+	return (*IVssBackupComponentsVTable)(unsafe.Pointer(vss.RawVTable))
+}
+
+// AbortBackup calls the equivalent VSS api.
+func (vss *IVssBackupComponents) AbortBackup() error {
+	result, _, _ := syscall.Syscall(vss.getVTable().abortBackup, 1,
+		uintptr(unsafe.Pointer(vss)), 0, 0)
+
+	return newVssErrorIfResultNotOK("AbortBackup() failed", HRESULT(result))
+}
+
+// InitializeForBackup calls the equivalent VSS api.
+func (vss *IVssBackupComponents) InitializeForBackup() error {
+	result, _, _ := syscall.Syscall(vss.getVTable().initializeForBackup, 2,
+		uintptr(unsafe.Pointer(vss)), 0, 0)
+
+	return newVssErrorIfResultNotOK("InitializeForBackup() failed", HRESULT(result))
+}
+
+// SetContext calls the equivalent VSS api.
+func (vss *IVssBackupComponents) SetContext(context VssContext) error {
+	result, _, _ := syscall.Syscall(vss.getVTable().setContext, 2,
+		uintptr(unsafe.Pointer(vss)), uintptr(context), 0)
+
+	return newVssErrorIfResultNotOK("SetContext() failed", HRESULT(result))
+}
+
+// GatherWriterMetadata calls the equivalent VSS api.
+func (vss *IVssBackupComponents) GatherWriterMetadata() (*IVSSAsync, error) {
+	var oleIUnknown *ole.IUnknown
+	result, _, _ := syscall.Syscall(vss.getVTable().gatherWriterMetadata, 2,
+		uintptr(unsafe.Pointer(vss)), uintptr(unsafe.Pointer(&oleIUnknown)), 0)
+
+	err := newVssErrorIfResultNotOK("GatherWriterMetadata() failed", HRESULT(result))
+	return vss.convertToVSSAsync(oleIUnknown, err)
+}
+
+// convertToVSSAsync looks up IVSSAsync interface if given result
+// is a success.
+func (vss *IVssBackupComponents) convertToVSSAsync(
+	oleIUnknown *ole.IUnknown, err error) (*IVSSAsync, error) {
+	if err != nil {
+		return nil, err
+	} else {
+		comInterface, err := queryInterface(oleIUnknown, UIID_IVSS_ASYNC)
+		if err != nil {
+			return nil, err
+		}
+
+		iVssAsync := (*IVSSAsync)(unsafe.Pointer(comInterface))
+		return iVssAsync, nil
+	}
+}
+
+// IsVolumeSupported calls the equivalent VSS api.
+func (vss *IVssBackupComponents) IsVolumeSupported(volumeName string) (bool, error) {
+	volumeNamePointer, err := syscall.UTF16PtrFromString(volumeName)
+	if err != nil {
+		panic(err)
+	}
+
+	isSupported := false
```

Comment:
You can't use `bool` here: see [this PR](https://github.com/WireGuard/wireguard-windows/commit/cdfdbae1bb2e392ef161532e56f0f29a73c43bce) for details.

Reply:
Fixed in https://github.com/restic/restic/pull/2274/commits/97322db98cdc31a9aba2c358e59809418cdf68be